test/parallel/test-http2-invalidheaderfield.js

```diff
@@ -0,0 +1,61 @@
+'use strict';
+const common = require('../common');
+if (!common.hasCrypto) { common.skip('missing crypto'); }
+
+// Check for:
+// Spaced headers
+// Psuedo headers
+// Capitalized headers
+
+const http2 = require('http2');
+const { throws } = require('assert');
+
+const server = http2.createServer(common.mustCall((req, res) => {
+
+  throws(() => {
+    res.setHeader(':path', '/');
+  }, TypeError);
+
+  throws(() => {
+    res.setHeader('t est', 123);
+  }, TypeError);
+
+  // should not fail
```

Comment:
Nit: you don't really need to write these, from the test itself it is understandable what's the expectation (wrapped in throws - should throw, not wrapped - must pass etc)

Reply:
The docs actually specify writing these.

https://nodejs.org/dist/latest-v14.x/docs/api/assert.html#assert_assert_doesnotthrow_fn_error_message

> Instead, consider adding a comment next to the specific code path that should not throw and keep error messages as expressive as possible.

Reply:
Well, that's if you want to use `doesNotThrow` which is understandable but we don't really use either `doesNotThrow` or the comment in the tests because then everything would be covered in those comments basically. Because throwing is never expected unless you specifically test for it with `throws`. That's why we usually just rely on unhandled/uncaught exception/rejection handlers.

Reply:
I see. I've made the changes.